src/main/java/com/dm/dmservicesandbox/domain/UserSignUpRequest.java

```diff
@@ -0,0 +1,73 @@
+package com.dm.dmservicesandbox.domain;
+
+import com.dm.dmservicesandbox.validation.PasswordMatches;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Size;
+
+
+@PasswordMatches
+@Getter
+@Setter
+public class UserSignUpRequest {
+
+    private Long userID;
+
+    private String userId;
+
+    @NotEmpty
+    private String displayName;
+
+    @NotEmpty
+    private String email;
+
+    @Size(min = 6, message = "{Size.userDto.password}")
+    private String password;
+
+    @NotEmpty
+    private String matchingPassword;
+
```

Comment:
We are missing also in this service the ability to pass the year of birth and the location... 
For location, we need to determine what we want to persist in db, what the user will choose is city, country, but we display on screen the City, region, country. Mapbox provides a mapbox id, but not sure if we can use this... the idea would be that we know exactly the location  for this without having to "search" for it as searches are not exact matches most of the time...

Reply:
also, I think we should persist that they accepted the terms and conditions

Reply:
yes its like a skeleton we will add more fields/ validations